src/components/Text.tsx

```diff
@@ -23,23 +30,15 @@ type TextProps = {
     family?: keyof typeof fontFamily;
 
     /** Any additional styles to apply */
-    style?: TextStyle | TextStyle[];
+    style?: StyleProp<TextStyle>;
 };
 
 function Text(
     {color = themeColors.text, fontSize = variables.fontSizeNormal, textAlign = 'left', children = null, family = 'EXP_NEUE', style = {}, ...props}: TextProps,
     ref: ForwardedRef<RNText>,
 ) {
     // If the style prop is an array of styles, we need to mix them all together
-    const mergedStyles = !Array.isArray(style)
-        ? style
-        : style.reduce(
-              (finalStyles, s) => ({
-                  ...finalStyles,
-                  ...s,
-              }),
-              {},
-          );
+    const mergedStyles = Array.isArray(style) ? Object.assign({}, ...lodashFlattenDeep(style as TextStyle[])) : style;
```

Comment:
How about this using StyleSheet instead of lodash?
```ts
    const componentStyle: TextStyle = {
        color,
        fontSize,
        textAlign,
        fontFamily: fontFamily[family],
        ...StyleSheet.flatten(style),
    };
```